src/cpp/src/whisper/context_tokens.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "context_tokens.hpp"
+
+namespace {
+std::pair<std::vector<int64_t>, float> tokenize(std::string&& text,
+                                                const ov::genai::WhisperGenerationConfig& config,
+                                                ov::genai::Tokenizer& tokenizer) {
+    if (text.empty()) {
+        return {{}, 0.0f};
+    }
+
+    auto start_time = std::chrono::steady_clock::now();
+    auto encoded = tokenizer.encode(text, ov::genai::add_special_tokens(false));
```

Comment:
```suggestion
    auto encoded = tokenizer.encode(std::move(text), ov::genai::add_special_tokens(false));
```
You copy otherwise. See https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:28,endLineNumber:7,positionColumn:28,positionLineNumber:7,selectionStartColumn:28,selectionStartLineNumber:7,startColumn:28,startLineNumber:7),source:'%23include+%3Ciostream%3E%0Astruct+A+%7B%0A++++A(const+A%26)+%7B%0A++++++++std::cout+%3C%3C+%22+copy%5Cn%22%3B%0A++++%7D%0A++++A(A%26%26)+%7B%0A++++++++std::cout+%3C%3C+%22+move%5Cn%22%3B%0A++++%7D%0A++++A()+%3D+default%3B%0A%7D%3B%0Avoid+encode(A)+%7B%7D%0Avoid+tokenize(A%26%26+a)+%7B%0A++++encode(a)%3B%0A%7D%0Aint+main()+%7B%0A++++tokenize(A%7B%7D)%3B%0A%7D%0A'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((g:!((h:compiler,i:(compiler:g142,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'0',intel:'0',libraryCode:'0',trim:'1',verboseDemangling:'0'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+gcc+14.2+(Editor+%231)',t:'0')),header:(),k:50,l:'4',m:50,n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compilerName:'x86-64+gcc+14.2',editorid:1,fontScale:14,fontUsePx:'0',j:1,wrap:'1'),l:'5',n:'0',o:'Output+of+x86-64+gcc+14.2+(Compiler+%231)',t:'0')),header:(),l:'4',m:50,n:'0',o:'',s:0,t:'0')),k:50,l:'3',n:'0',o:'',t:'0')),l:'2',n:'0',o:'',t:'0')),version:4

Reply:
Thanks for example! I simple update to use `std::move` doesn't work as there is no appropriate `encode` overload.
Compilation succeeds with the following changes:
```c++
  TokenizedInputs encode(const std::string prompt, const ov::AnyMap& tokenization_params = {});
```
->
```c++
  TokenizedInputs encode(const std::string& prompt, const ov::AnyMap& tokenization_params = {});
  TokenizedInputs encode(const std::string&& prompt, const ov::AnyMap& tokenization_params = {});
```

If it's critical to use `std::move` here, I propose to address it in another PR.

Reply:
No, it's not critical